packages/fiber/src/core/store.ts

```diff
@@ -217,6 +215,38 @@ const createStore = (
     const setPerformanceCurrent = (current: number) =>
       set((state) => ({ performance: { ...state.performance, current } }))
 
+    // Handle frame behavior in WebXR
+    const handleXRFrame = (timestamp: number) => {
+      const state = get()
+      if (state.frameloop === 'never') return
+
+      advance(timestamp, true)
+    }
+
+    // Toggle render switching on session
+    const handleSessionChange = () => {
+      gl.xr.enabled = gl.xr.isPresenting
+      gl.setAnimationLoop(gl.xr.isPresenting ? handleXRFrame : null)
+
+      // If exiting session, request frame
+      if (!gl.xr.isPresenting) invalidate(get())
+    }
+
+    // WebXR session manager
+    const xr = {
+      connect() {
+        gl.xr.addEventListener('sessionstart', handleSessionChange)
+        gl.xr.addEventListener('sessionend', handleSessionChange)
+      },
+      disconnect() {
+        gl.xr.removeEventListener('sessionstart', handleSessionChange)
+        gl.xr.removeEventListener('sessionend', handleSessionChange)
+      },
+    }
```

Comment:
Should this go in `web/events`?

Reply:
No I don't think so, I'd say `events` is for user inputs.

Reply:
That's my reservation, too. Just wary of including environment-specific features in the core.